core/src/solve.jl

```diff
@@ -11,6 +11,7 @@ node_id: All the IDs of the nodes that are in this subnetwork
 node_id_mapping: Mapping Dictionary; model_node_id => AG_node_id where such a correspondence exists
     (all AG node ids are in the values)
 node_id_mapping_inverse: The inverse of node_id_mapping, Dictionary; AG node ID => model node ID
+allocgraph_edge_ids_user_demand: AG user node ID => AG user inflow edge ID
```

Comment:
Perhaps we should consider using https://github.com/JuliaGraphs/MetaGraphsNext.jl and only use IDs, to avoid having to switch from ID to vertex index all the time, also for the flow graph, related to https://github.com/Deltares/Ribasim/issues/694.

Not to be addressed in this PR, but curious to hear your thoughts.

Reply:
The IDs are indeed different (for one because a graph from `Graph` requires consecutive node IDs). Using `MetaGraphsNext.nl` could certainly be very interesting, now translating between identifiers is quite a hassle.